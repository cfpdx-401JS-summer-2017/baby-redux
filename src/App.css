.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

#greeter {margin: 1em 0;}
#greeting {
  font-weight: bold;
  color: #66b;
  font-size: 1.5em;
  margin-bottom: 1em;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
