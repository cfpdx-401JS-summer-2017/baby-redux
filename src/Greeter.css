label {
    margin-bottom: 10px;
    font-weight: bold;
}

input {
    margin-bottom: 10px;
}

.salutation {
    font-family: 'Luckiest Guy', cursive;
    font-size: 30px;
    margin-bottom: 50px;
}

.input {
    display: flex;
    flex-direction: column;
    width: 200px;
    margin: 0 auto;
}

button {
    width: 70px;
    height: 30px;
    font-size: 75%;
    background-color: black;
    color: white;
    text-transform: uppercase;
    border-radius: 5px;
    border: none;
    margin: 0 auto;
    margin-top: 20px;
}

button:focus{
    outline: none;
}