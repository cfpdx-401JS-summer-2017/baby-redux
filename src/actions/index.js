import { ADD_GREETING, ADD_NAME, RESET } from '../containers/salutation.constants';

export const addGreeting = (payload) => ({
    type: ADD_GREETING,
    payload
});

export const addName = (payload) => ({
    type: ADD_NAME,
    payload
});

export const reset = () => ({
    type:RESET
});