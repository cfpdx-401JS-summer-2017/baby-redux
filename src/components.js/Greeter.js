import React from 'react';
import PropTypes from 'prop-types';

function Greeter() {
  Greeter.propTypes = {
    salutation: PropTypes.string,
    name: PropTypes.string
  }


  return (
    <div>
      <span>{salutation}</span>
      <span>{name}</span>!
    </div>
  );
}

export default Greeter;
