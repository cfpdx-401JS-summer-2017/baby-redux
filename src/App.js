import React, { Component } from 'react';
import Greeter from './Greeter';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <div className="App-header">
          <h2>Hey pal, go greet yourself</h2>
        </div>
        <div>
          <Greeter/>
        </div>
      </div>
    );
  }
}

export default App;
