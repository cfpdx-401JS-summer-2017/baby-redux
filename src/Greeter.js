import React from 'react';
import {connect} from 'react-redux';
import {updateSalutation, updateName, reset} from './greeter.actions';

export function Greeter({salutation = '', name = '', dispatch}) {
    return (
        <div>
            <span>{salutation}, </span>
            <span>{name}</span>!
            <p>Salutation: </p>
            <input value={salutation} onChange={({target}) => dispatch(updateSalutation(target.value))}></input>
            <p>Name: </p>
            <input value={name} onChange={({target}) => dispatch(updateName(target.value))}></input>
            <br/>
            <button onClick={() => dispatch(reset())}>Reset</button>
        </div>
    )
}

export default connect(
    state => ({salutation: state.salutation, name: state.name})
)(Greeter);