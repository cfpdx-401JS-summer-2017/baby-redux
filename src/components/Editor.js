import React from 'react';
import PropTypes from 'prop-types';
import { updateSalutation, updateName, resetForm } from '../greeter.actions';

export default function Editor({ salutation, name, dispatch }) {
  Editor.propTypes = {
    salutation: PropTypes.string,
    name: PropTypes.string,
    dispatch: PropTypes.func
  }

  return (
    <div className="greetForm">
      <input
        type="text"
        placeholder="salutation"
        value={salutation}
        onChange={({ target }) =>
          dispatch(updateSalutation({ salutation: target.value, name: name }))}
      />
      <input
        type="text"
        placeholder="name"
        value={name}
        onChange={({ target }) =>
          dispatch(updateName({ salutation: salutation, name: target.value }))}
      />
      <button onClick={() => dispatch(resetForm())} >reset form</button>
    </div>
  );
}
