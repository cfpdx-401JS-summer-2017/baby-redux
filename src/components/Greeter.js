import React from 'react';
import { connect } from 'react-redux';
import { Editor } from './Editor';

export function Greeter({salutation, name, dispatch}) {
  console.log(salutation, name);
  return (
    <div>
      <div>
        <span>{salutation}</span>
        <span>{name}</span>!
      </div>
      <Editor salutation={salutation} name={name} dispatch={dispatch}/>
    </div>
  );
}

export default connect(
  state => ({
  salutation: state.salutation,
  name: state.name
}))(Greeter);
