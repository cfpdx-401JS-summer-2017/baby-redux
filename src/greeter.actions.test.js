import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import App from './App';
import Greeter from './components/Greeter';
import Editor from './components/Editor';
import { updateSalutation, updateName, resetForm } from './greeter.actions';

// action creators & reducers

describe('greeter', () => {

  describe('actions', () => {


  console.log(this);
  // const wrapper = shallow(<Counter count={3}/>);
  // expect(toJSON(wrapper)).toMatchSnapshot();

  it('updates salutation', () =>{
    const wrapper = mount
    (<Editor onChange={updateSalutation({salutation: 'hello', name: 'world'})} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
    const wrapperAfterChange = mount
    (<Editor onChange={updateSalutation({salutation: 'goodbye', name: 'stranger'})} />);
    expect(toJSON(wrapperAfterChange)).toMatchSnapshot();
  });

  it('updates name', () => {
    const wrapper = mount
    (<Editor onChange={updateName({salutation: 'hello', name: 'sailor'})} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
    const wrapperAfterChange = mount
    (<Editor onChange={updateName({salutation: 'hello', name: 'Larry'})} />);
    expect(toJSON(wrapperAfterChange)).toMatchSnapshot();

  });

  it('resets the form', () =>{

  });

  it('updates the state', () =>{

  });

})

})
