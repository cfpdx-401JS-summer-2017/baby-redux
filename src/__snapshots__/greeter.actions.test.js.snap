// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`greeter actions updates name 1`] = `
<Editor
  onChange={
    Object {
      "payload": Object {
        "name": "sailor",
        "salutation": "hello",
      },
      "type": "CHANGE",
    }
  }
>
  <div
    className="greetForm"
  >
    <input
      onChange={[Function]}
      placeholder="salutation"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="name"
      type="text"
    />
    <button
      onClick={[Function]}
    >
      reset form
    </button>
  </div>
</Editor>
`;

exports[`greeter actions updates name 2`] = `
<Editor
  onChange={
    Object {
      "payload": Object {
        "name": "Larry",
        "salutation": "hello",
      },
      "type": "CHANGE",
    }
  }
>
  <div
    className="greetForm"
  >
    <input
      onChange={[Function]}
      placeholder="salutation"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="name"
      type="text"
    />
    <button
      onClick={[Function]}
    >
      reset form
    </button>
  </div>
</Editor>
`;

exports[`greeter actions updates salutation 1`] = `
<Editor
  onChange={
    Object {
      "payload": Object {
        "name": "world",
        "salutation": "hello",
      },
      "type": "CHANGE",
    }
  }
>
  <div
    className="greetForm"
  >
    <input
      onChange={[Function]}
      placeholder="salutation"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="name"
      type="text"
    />
    <button
      onClick={[Function]}
    >
      reset form
    </button>
  </div>
</Editor>
`;

exports[`greeter actions updates salutation 2`] = `
<Editor
  onChange={
    Object {
      "payload": Object {
        "name": "stranger",
        "salutation": "goodbye",
      },
      "type": "CHANGE",
    }
  }
>
  <div
    className="greetForm"
  >
    <input
      onChange={[Function]}
      placeholder="salutation"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="name"
      type="text"
    />
    <button
      onClick={[Function]}
    >
      reset form
    </button>
  </div>
</Editor>
`;
